railties/lib/rails/test_unit/line_filtering.rb

```diff
@@ -1,78 +1,12 @@
 require "method_source"
+require "rails/test_unit/runner"
 
 module Rails
   module LineFiltering # :nodoc:
     def run(reporter, options = {})
-      if options[:patterns] && options[:patterns].any? { |p| p =~ /:\d+/ }
-        options[:filter] = \
-          CompositeFilter.new(self, options[:filter], options[:patterns])
-      end
+      options[:filter] = Rails::TestUnit::Runner.compose_filter(self, options[:filter])
```

Comment:
In a later PR, I'd like to follow minitest-focus'es lead to make the filters regexps. I.e. move the filter definitions into `plugin_rails_init` by resolving line numbers to a specific method name.

Needs to handle a file creating multiple Runnables though and running with multiple line numbers (test/models/post_test.rb:4:8). Tried to do it for this, but faced way too much head-desking.